Tests/Basic/StringConversionsTests.swift

```diff
@@ -0,0 +1,44 @@
+/*
+This source file is part of the Swift.org open source project
+
+Copyright 2016 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See http://swift.org/LICENSE.txt for license information
+See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+import Basic
+
+class ShellEscapedTests: XCTestCase {
+
+    func testShellEscaped() {
+
+        var str = "hello"
+        XCTAssertEqual("hello", str.shellEscaped())
+
+        str = "hello world"
+        XCTAssertEqual("'hello world'", str.shellEscaped())
+
+        str = "hello 'world"
+        str.shellEscape()
+        XCTAssertEqual("'hello \\'world'", str)
```

Comment:
This isn't correct, at least not for Bash, Zsh, and Csh (on macOS, at least).

Doing `echo 'hello \'world'` yields an unterminated-quote error.  This actually needs to be `'hello '\''world'`.